src/EditorFeatures/CSharpTest/MakeLocalFunctionStatic/PassInCapturedVariablesAsArgumentsCodeFixProviderTests.cs

```diff
@@ -0,0 +1,347 @@
+        public async Task TestRecursiveCall()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int N(int x)
+    {
+        int y = 10;
+        var m = AddLocal(1, 2);
+        return AddLocal(m, m);
+
+        static int AddLocal(int a, int b)
+        {
+            return AddLocal(a, b) + [||]x + y;
+        }
+    }
+}",
+@"class C
+{
+    int N(int x)
+    {
+        int y = 10;
+        var m = AddLocal(1, 2, x, y);
+        return AddLocal(m, m, x, y);
+
+        static int AddLocal(int a, int b, int x, int y)
+        {
+            return AddLocal(a, b, x, y) + x + y;
+        }
+    }
+}", parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        public async Task TestCallInArgumentList()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int N(int x)
+    {
+        int y = 10;
+        return AddLocal(AddLocal(1, 2), AddLocal(3, 4));
+
+        static int AddLocal(int a, int b)
+        {
+            return AddLocal(a, b) + [||]x + y;
+        }
+    }
+}",
+@"class C
+{
+    int N(int x)
+    {
+        int y = 10;
+        return AddLocal(AddLocal(1, 2, x, y), AddLocal(3, 4, x, y), x, y);
+
+        static int AddLocal(int a, int b, int x, int y)
+        {
+            return AddLocal(a, b, x, y) + x + y;
+        }
+    }
+}", parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        public async Task TestCallsWithNamedArguments()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int N(int x)
+    {
+        int y = 10;
+        var m = AddLocal(1, b: 2);
+        return AddLocal(b: m, a: m);
+
+        static int AddLocal(int a, int b)
+        {
+            return a + b + [||]x + y;
+        }
+    }
+}",
+@"class C
+{
+    int N(int x)
+    {
+        int y = 10;
+        var m = AddLocal(1, b: 2, x: x, y: y);
+        return AddLocal(b: m, a: m, x: x, y: y);
+
+        static int AddLocal(int a, int b, int x, int y)
+        {
+            return a + b + x + y;
+        }
+    }
+}"
+, parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        public async Task TestCallsWithDafaultValue()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int N(int x)
+    {
+        string y = "";
+        var m = AddLocal(1);
+        return AddLocal(b: m);
+
+        static int AddLocal(int a = 0, int b = 0)
+        {
+            return a + b + x + [||]y.Length;
+        }
+    }
+}",
+@"class C
+{
+    int N(int x)
+    {
+        string y = "";
+        var m = AddLocal(1, x: x, y: y);
+        return AddLocal(b: m, x: x, y: y);
+
+        static int AddLocal(int a = 0, int b = 0, int x = 0, string y = null)
```

Comment:
>int x = 0, string y = null [](start = 50, length = 26)

Wondering if this would be the user preferred behavior or whether they would prefer the new arguments being non-optional.

Reply:
This is implemented this way mostly for simplicity, I think it's fairly easy for user to adjust this after the refactoring is done. Besides, this is for local function, so I don't expect optional parameter to be very common.

I will keep this as is for now.